src/rules/ec2/EC2EBSVolumeEncrypted.ts

```diff
@@ -24,25 +24,19 @@ export default Object.defineProperty(
       }
       return NagRuleCompliance.COMPLIANT;
     } else if (node instanceof CfnInstance) {
-      const blockDeviceMappings = Stack.of(node).resolve(
-        node.blockDeviceMappings
-      );
-      if (blockDeviceMappings !== undefined) {
-        for (const blockDeviceMapping of blockDeviceMappings) {
-          const encryption = NagRules.resolveIfPrimitive(
-            node,
-            blockDeviceMapping.ebs.encrypted
-          );
-          if (encryption !== true) {
-            return NagRuleCompliance.NON_COMPLIANT;
-          }
-        }
-        return NagRuleCompliance.COMPLIANT;
+      const instanceEBSCompliant = isInstanceEBSCompliant(node);
+      const launchTemplateCompliant = isLaunchTemplateCompliant(node);
+
+      if (
+        instanceEBSCompliant === undefined &&
+        launchTemplateCompliant === undefined
+      ) {
+        return NagRuleCompliance.NON_COMPLIANT;
       }
-      if (isLaunchTemplateCompliant(node)) {
-        return NagRuleCompliance.COMPLIANT;
+      if (instanceEBSCompliant === false || launchTemplateCompliant === false) {
+        return NagRuleCompliance.NON_COMPLIANT;
```

Comment:
I understand what is happening here, but it is also very confusing to read. The functions `isInstanceEBSCompliant` and   `isLaunchTemplateCompliant` aren't doing what they look like they should be doing 

1.  Rename `isInstanceEBSCompliant` to `InstanceEBSState` and `isLaunchTemplateCompliant` to `InstanceLaunchTemplateState`
2.  Return an enum for `BlockDevicesState` instead of returning a `boolean` or `undefined`
    - ```ts
       enum BlockDevicesState {
         Encrypted,
         Unencrypted,
         Absent
       }
      ```

Reply:
Thank you for your suggestion! Fixed!